src/Microsoft.AspNet.Mvc.Core/RedirectToActionResult.cs

```diff
@@ -39,7 +39,7 @@ public class RedirectToActionResult : ActionResult, IKeepTempDataResult
 
         public string ControllerName { get; set; }
 
-        public IDictionary<string, object> RouteValues { get; set; }
+        public RouteValueDictionary RouteValues { get; set; }
```

Comment:
If you're feel overtly enthusiastic, copy the description from the other `*Result`s